### PR TITLE
Primitive call

### LINKED ISSUE #460: Improve API for calling primitive functions
We should make it easier for a user to call primitive functions directly.

Currently, a user has to initialize the primitive and then call `primitve.get_function()`. 

```
In [1]: from featuretools.primitives import NMostCommon 
   ...: import pandas as pd 
   ...: most_common_3 = NMostCommon(n=3) 
   ...: values = pd.Series(["a", "b", "a", "b", "c", "c", "d"]) 
   ...: func = most_common_3.get_function()                                
   ...: func(values)                                                                                                                                                                                             
Out[1]: array(['c', 'b', 'a'], dtype=object)
```
The proposal is to make it like this

```
In [1]: from featuretools.primitives import NMostCommon 
   ...: import pandas as pd 
   ...: most_common_3 = NMostCommon(n=3) 
   ...: values = pd.Series(["a", "b", "a", "b", "c", "c", "d"]) 
   ...: most_common_3(values)                                                                                                                                                                                             
Out[1]: array(['c', 'b', 'a'], dtype=object)
```

### PR DESCRIPTION
**Pull request by CharlesBradshaw**
We don't want to call get_function in the init.

We could store the output of self.get_function() and check if that exists every time in the call function. 
``` python
def __call__(self,data):
    if not hasattr(self,method):
        self.method = self.get_function()
    return self.method(data)
```
That goes against the pythonic "it's Easier to Ask for Forgiveness than Permission" (EAFP) *in this case we're using "Look Before You  Leap" (LBYL)*. To fix this we can use a try except instead.

```python
def __call__(self,data):
    try:
        return self.method(data)
    except AttributeError:
        self.method = self.get_function()
        return self.method(data)
```
We already know when we need to call `get_function` , and it's on the first time` __call__` is run. So instead of using the try except function we can call `get_function` in the `__call__` function, and then redefine `__call__` to be the output of `get_function` This way there are no unnecessary checks

```python
def __call__(self,data):
    self.__call__ = self.get_function()
    return self.__call__(data)
```

This approach doesn't work because `__call__` is a special  method. When a class instance, for example `primitive` gets called like `primitive(data)`, python under the hood runs `type(primitive).__call__(primitive,data)` instead of `primitive.__call__data`

We can get around this by using a non special method as an intermediate.

```python
def __call__(self,*args,**kwargs):
    self._temp_call(*args,**kwargs)

def _temp_call(self, *args, **kwargs):
    self._temp_call = self.get_function().__get__(self)
    return self._temp_call(*args,**kwargs)
```
This way the second time `_temp_call` has been called, it's set to the output of `get_function`. 

Note that the `self.get_function()` has `__get__(self)` also. This is because to add a function to an instance of a class, and not the class itself, you have to bind the instance to the function, aka add `self` as the first parameter always. I don't fully understand how this works, but it's done using `__get__(x)`

Given that we can't overwrite __call__, and that it isn't standard to add/update the methods of an instance, it feels like the if statement, or the try except would be the better solution

Fixes #460

### CODE REVIEW

File: featuretools/tests/primitive_tests/test_primitive_base.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from featuretools.primitives import Max
+
+
+def test_call():
```
Comment (kmax12) [inline featuretools/tests/primitive_tests/test_primitive_base.py]: can you add a test where the primitive takes multiple arguments?

File: featuretools/primitives/base/primitive_base.py [hunk 2]
```diff
@@ -34,6 +34,14 @@ class PrimitiveBase(object):
     # (bool) If True will only make one feature per unique set of base features
     commutative = False
 
+    def __call__(self, data):
+        # We only want to call get_function once, when __call__ is run for the
+        # first time. Since we know that we don't need to call get_function after
+        # that, we can redefine the __call__ function instead of checking for if
+        # it is the first time __call__ has been called
+        self.__call__ = self.get_function()
```
Comment (kmax12) [inline featuretools/primitives/base/primitive_base.py]: it's a bit weird to overwrite the method at run time. we can just do 

```
if self.primitive_function is None:
    self.primitive_function = self.get_function()

return  self.primitive_function(data)
```

File: featuretools/primitives/base/primitive_base.py [hunk 3]
```diff
@@ -34,6 +34,14 @@ class PrimitiveBase(object):
     # (bool) If True will only make one feature per unique set of base features
     commutative = False
 
+    def __call__(self, data):
```
Comment (kmax12) [inline featuretools/primitives/base/primitive_base.py]: this won't handle primitives that requires to input variables. We should use `*args` and `**kwargs`

File: featuretools/tests/primitive_tests/test_primitive_base.py (hunk 1 above)
Comment (kmax12) [reply featuretools/tests/primitive_tests/test_primitive_base.py]: add a test for both an aggregation and transform primitive

Comment (kmax12) [reply featuretools/tests/primitive_tests/test_primitive_base.py]: add a test for primitives that have `uses_calc_time=True`  and get a `time` argument

File: featuretools/tests/primitive_tests/test_primitive_base.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    answer = 86400.0
+    assert answer == primitive(datetimes, time=datetime(2015, 6, 8))
+
+
+def test_call_multiple_args():
+    class TestPrimitive(PrimitiveBase):
+        def get_function(self):
+            def test(x, y):
+                print("x", x)
+                print("y", y)
```
Comment (CharlesBradshaw) [inline featuretools/tests/primitive_tests/test_primitive_base.py]: Remove prints

**Pull request merged**